packages/nodes-base/nodes/Xero/GenericFunctions.ts

```diff
@@ -12,13 +12,21 @@ import {
 	IDataObject,
 } from 'n8n-workflow';
 
+import * as _ from 'lodash';
+
 export async function xeroApiRequest(this: IExecuteFunctions | IExecuteSingleFunctions | ILoadOptionsFunctions, method: string, resource: string, body: any = {}, qs: IDataObject = {}, uri?: string, headers: IDataObject = {}): Promise<any> { // tslint:disable-line:no-any
+	// when this function is called in loops with multiple items, deleting the organizationId from the body
+	// causes authentication errors, but keeping it in the body causes errors when creating a contact (or other such one-off requests).
+	// this ensures the organizationId is preserved, and also kept out of the body of the request
+	const bodyCopy = _.cloneDeep(body);
+	delete bodyCopy.organizationId;
```

Comment:
I don't think you need a deep clone here, and you should use built-in methods over libraries.

```suggestion
	const { organizationId, ...bodyCopy } = body;
```

Reply:
Updated